src/methods/sample.jl

```diff
@@ -0,0 +1,61 @@
+#=
+# Sample
+
+```@example sample
+import GeoInterface as GI, GeometryOps as GO
+p1 = GI.Polygon([[[-55965.680060140774, -31588.16072168928], [-55956.50771556479, -31478.09258677756], [-31577.548550575284, -6897.015828572996], [-15286.184961223798, -15386.952072224134], [-9074.387601621409, -27468.20712382156], [-8183.4538916097845, -31040.003969070774], [-27011.85123029944, -38229.02388009402], [-54954.72822634951, -32258.9734800704], [-55965.680060140774, -31588.16072168928]]])
+points = GO.sample(p1, 100)
+using CairoMakie
+f, a, p = poly(p1)
+scatter!(a, points)
+f
+```
+
+=#
+
+export sample, UniformSampling
+
+struct UniformSampling
+end
+
+application_level(::UniformSampling) = TraitTarget(GI.MultiPolygonTrait(), GI.MultiLineStringTrait(), GI.MultiPointTrait(), GI.PolygonTrait(), GI.LineStringTrait())
+
+function sample(geom, n::Int)
+    return sample(UniformSampling(), geom, n)
+end
+
+function sample(alg, geom, n)
+    return apply(x -> _sample(alg, GI.trait(x), x, n), application_level(alg), geom)
```

Comment:
At this point each feature will be sampled independently, so it's a dataframe to dataframe transfer.  Since everything else also works the same way (centroid etc) I think this is at least consistent?

Am open to discussion though, not sure how people generally use this stuff.

Reply:
GeoPandas samples per geometry, and returns a multipoint:
https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.sample_points.html

`GO.centroid` does return a single point even if you pass in a feature collection or table.  (There's an issue with the Natural Earth north Korea geometry as well :D)

Reply:
Aah, yeah apply won't work here since it will try to reconstruct.

I guess this is a good time to figure out the generic apply iterator thing?